components/brave_rewards/browser/database/database_recurring_tip.cc

```diff
@@ -0,0 +1,183 @@
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include <string>
+#include <utility>
+
+#include "base/bind.h"
+#include "base/strings/stringprintf.h"
+#include "brave/components/brave_rewards/browser/database/database_recurring_tip.h"
+#include "sql/statement.h"
+#include "sql/transaction.h"
+
+namespace brave_rewards {
+
+namespace {
+  // TODO(https://github.com/brave/brave-browser/issues/7144):
+  //  rename to recurring_tip
+  const char* table_name_ = "recurring_donation";
+  const int minimum_version_ = 2;
+}  // namespace
+
+DatabaseRecurringTip::DatabaseRecurringTip(
+    int current_db_version) :
+    DatabaseTable(current_db_version) {
+}
+
+DatabaseRecurringTip::~DatabaseRecurringTip() = default;
+
+bool DatabaseRecurringTip::Init(sql::Database* db) {
+  if (GetCurrentDBVersion() < minimum_version_) {
+    return true;
+  }
+
+  sql::Transaction transaction(db);
+  if (!transaction.Begin()) {
+    return false;
+  }
+
+  bool success = CreateTable(db);
+  if (!success) {
+    return false;
+  }
+
+  success = CreateIndex(db);
+  if (!success) {
+    return false;
+  }
+
+  return transaction.Commit();
+}
+
+bool DatabaseRecurringTip::CreateTable(sql::Database* db) {
+  if (db->DoesTableExist(table_name_)) {
```

Comment:
I know this is probably just restructured/relocated code - but can we start checking 'db' before using it as part of this PR?

Reply:
will be fixed here https://github.com/brave/brave-browser/issues/7145